pkg/operator/ceohelpers/common.go

```diff
@@ -127,3 +129,36 @@ func memberToURL(member *etcdserverpb.Member) (string, error) {
 	}
 	return member.PeerURLs[0], nil
 }
+
+// FindMachineByNodeInternalIP finds the machine that matches the given nodeInternalIP
+// is safe because the MAO:
+//  syncs the addresses in the Machine with those assigned to real nodes by the cloud provider,
+//  checks that the Machine and Node lists match before issuing a serving certification for the kubelet
+//  when the host disappears from the cloud side, it stops updating the Machine so the addresses and information
+//  should persist there as a tombstone as the Machine is marked Failed
+func FindMachineByNodeInternalIP(nodeInternalIP string, machineSelector labels.Selector, machineLister machinelistersv1beta1.MachineLister) (*machinev1beta1.Machine, error) {
```

Comment:
we could create a new PR, I think that would make @deads2k happy and result in smaller diff

Reply:
ptal, https://github.com/openshift/cluster-etcd-operator/pull/795